app/main.py

```diff
@@ -1,6 +1,19 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_store = {}
+
+    def wrapper(*args, **kwargs) -> Any:
+        key = (args, frozenset(kwargs.items()))
+
+        if key in cache_store:
+            print("Getting from cache")
+            return cache_store[key]
+        else:
+            result = func(*args, **kwargs)
+            cache_store[key] = result
+            print("Calculating new result")
+            return result
```

Comment:
Rewrite this function with using `return` statement one time

Reply:
Okay, boss!